Logger leftovers

dsvoronin opened the pull request:
see commit messages for changes details

dsvoronin committed:
single HttpElasticClient instance

dsvoronin committed:
log in timeprovider

dsvoronin committed:
test with one symbol day and month

dsvoronin committed:
RetryInterceptor now capable of modifying request before retry

ElasticClient now uses roundRobin on call retries

dsvoronin committed:
disable RedundantVisibilityModifierRule because of kotlin explicit api mode

dsvoronin committed:
fix compile error

eugene-krivobokov commented on subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/ElasticClientFactory.kt:
```diff
@@ -0,0 +1,43 @@
+
+    private const val defaultTimeoutSec = 10L
+
+    private val timeProvider: TimeProvider by lazy { DefaultTimeProvider() }
```
Is it so expensive to make it lazy?

dsvoronin replied:
nope, in fact it maybe more expensive to make it lazy here; i'll fix it

eugene-krivobokov commented on subprojects/common/time/src/main/kotlin/com/avito/time/DefaultTimeProvider.kt:
```diff
@@ -12,9 +14,17 @@ import java.util.concurrent.TimeUnit
+    override val timeZone: TimeZone = TimeZone.getDefault()
+
+    init {
+        logger.debug("Creating new time provider: Locale = $locale; TimeZone = $timeZone")
```
Even debug seems too verbose. Time provider is a lightweight interface about standard classes: TimeZone, Locale, DateFormat, ... Don't see reasons to log it.

dsvoronin replied:
It's in fact crucial for this PR; i want to log defaults for locale and timezone for all environments: gradle, emulators

eugene-krivobokov replied:
> i want to log defaults for locale and timezone for all environments

Ok, that seems reasonable. I propose to do it one time while starting the environment. 
I found it excessive to log the environment on each creation of this simple util.

The reason is simple. If we want to check the environment, it will be good to do it explicitly and as early as possible.

eugene-krivobokov commented on subprojects/common/okhttp/src/main/kotlin/com/avito/http/RetryInterceptor.kt:
```diff
@@ -42,11 +44,17 @@ class RetryInterceptor constructor(
 
             try {
                 prepareForRetry(response)
+                val request = if (tryCount > 1) {
```
Do we modify only the second retry? Why?

dsvoronin replied:
on second try; we don't modify only original request processing

eugene-krivobokov replied:
Sorry, missed that we do all `proceed` in the loop.

eugene-krivobokov commented on subprojects/detekt.yml:
```diff
@@ -675,8 +675,10 @@ style:
         active: false
     # https://detekt.github.io/detekt/style.html#redundantvisibilitymodifierrule
+    # todo don't know about kotlin strict mode
+    # possibly fixed in 1.15 https://github.com/detekt/detekt/issues/3125
```
Can we just update to 1.15?

dsvoronin replied:
yep, i thought it would be better in separate pr but whatever

dsvoronin replied:
fix in 1.15 https://github.com/detekt/detekt/issues/3125 only works per module, not in our detektAll task
because of how strict api detection works

dsvoronin committed:
make ElasticClientFactory vals not lazy

describe RedundantVisibilityModifierRule problem

update detekt to release version

dsvoronin committed:
trying to fix TZ dependent test
